app/src/org/odk/collect/android/jr/extensions/IntentCallout.java

```diff
@@ -114,68 +112,74 @@ public boolean processResponse(Intent intent, ODKView currentView, FormInstance
-            EvaluationContext context = new EvaluationContext(form.getEvaluationContext(), ref);
-
-            AbstractTreeElement node = context.resolveReference(ref);
-            
-            if(node == null) {
-                //continue?
-                
-            }
-            int dataType = node.getDataType();
-            
-            //TODO: Handle file system errors in a way that is more visible to the user
-            
-            //See if this is binary data and we'll have to do something complex...
-            if(dataType == Constants.DATATYPE_BINARY) {
-                //We need to copy the binary data at this address into the appropriate location
-                if(responseValue == null || responseValue.equals("")) {
-                    //If the response was blank, wipe out any data that was present before
-                    form.setValue(null, ref);
+        if(responses != null) {
+            for (String key : responses.keySet()) {
+                //See if the value exists at all, if not, skip it
+                if (!response.containsKey(key)) {
                     continue;
                 }
-                
-                //Otherwise, grab that file
-                File src = new File(responseValue);
-                if(!src.exists()) {
-                    //TODO: How hard should we be failing here?
-                    Log.w("FormEntryActivity-Callout", "ODK received a link to a file at " + src.toString() + " to be included in the form, but it was not present on the phone!");
-                    //Wipe out any reference that exists
-                    form.setValue(null, ref);
-                    continue;
+
+                //Get our response value
+                String responseValue = response.getString(key);
+                if (key == null) {
+                    key = "";
                 }
-                
-                File newFile = new File(destination, src.getName());
-                
-                //Looks like our source file exists, so let's go grab it
-                FileUtils.copyFile(src, newFile);
-                
-                //That code throws no errors, so we have to manually check whether the copy worked.
-                if(newFile.exists() && newFile.length() == src.length()) {
-                    form.setValue(new StringData(newFile.toString()), ref);
-                    continue;
-                } else {
-                    Log.e("FormEntryActivity-Callout", "ODK Failed to property write a file to " + newFile.toString());
-                    form.setValue(null, ref);
-                    continue;    
+
+                //Figure out where it's going
+                TreeReference ref = responses.get(key);
+
+                EvaluationContext context = new EvaluationContext(form.getEvaluationContext(), ref);
+
+                AbstractTreeElement node = context.resolveReference(ref);
+
+                if (node == null) {
+                    //continue?
```

Comment:
You probably want a 'continue' here, otherwise the next line will cause a crash.